src/objects/Mesh.js

```diff
@@ -375,7 +380,23 @@ function checkBufferGeometryIntersection( object, material, raycaster, ray, uv,
 			_uvB.fromBufferAttribute( uv2, b );
 			_uvC.fromBufferAttribute( uv2, c );
 
-			intersection.uv2 = Triangle.getUV( _intersectionPoint, _vA, _vB, _vC, _uvA, _uvB, _uvC, new Vector2() );
+			intersection.uv2 = Triangle.getInterpolation( _intersectionPoint, _vA, _vB, _vC, _uvA, _uvB, _uvC, new Vector2() );
+
+		}
+
+		if ( normal ) {
```

Comment:
Since normals are often present in geometries, this code will be execute often as well although the interpolated normal won't be required in most apps.

I wonder if we should make the query/computation optional by adding a configuration option to `Raycaster` (and maybe uv and uv2, too). On the other hand, the computation is only done per intersection point so it shouldn't produce much overhead.

Reply:
I considered this one (and I’ll do it if you think it’s necessary), but came to the same conclusion that it’s out of the hot-path and a relatively minor additional computation…

However, it might be nice to control how much garbage/allocation raycaster intersections make…. Perhaps either by a config option or by feeding a pool of results to override…

I will say though, that coming from Unity, not having normals on the top-level intersection object confused me for a good 20 minutes the first time I worked with Raycaster… It’d be nice to have it by default :-)

Reply:
I'm okay with keeping the code like it is for now. If someone complains about performance, we can still add a configuration option in `Raycaster.params`.